sdlib/d/gc/tcache.d

```diff
@@ -43,12 +44,11 @@ public:
 			auto ptr = arena.allocSmall(emap, asize);
 			auto pd = getPageDescriptor(ptr);
 			auto si = SlabAllocInfo(pd, ptr);
-			si.setUsedCapacity(size);
-			assert(finalizer is null, "finalizer not yet supported for slab!");
+			si.initializeMetadata(finalizer, size);
 			return ptr;
 		}
 
-		auto ptr = arena.allocLarge(emap, asize, false);
+		auto ptr = arena.allocLarge(emap, getAllocSize(size), false);
```

Comment:
We can extend large allocations in place, so we might want to go for the proper size, no?